server/memo.go

```diff
@@ -702,3 +710,49 @@ func (s *Server) getMemoDisplayWithUpdatedTsSettingValue(ctx context.Context) (b
 	}
 	return memoDisplayWithUpdatedTs, nil
 }
+
+// FilterPublicMemosByDays filter public MemoMessages by creator's public-in-days setting.
+func FilterPublicMemosByDays(ctx context.Context, s *store.Store, list []*store.MemoMessage) ([]*store.MemoMessage, error) {
+	settings, err := s.FindUserSettingList(ctx, &api.UserSettingFind{Key: api.UserSettingPublicInDays})
```

Comment:
`FindUserSetting` will be enough.

Reply:
We need fetch setting for each creator, not only the current user. So FindUserSetting is not satisfied.